### PR TITLE
feat(tools): add sentry to client for error handling

### LINKED ISSUE #39751: Client Sentry integration
**Is your feature request related to a problem? Please describe.**
Currently, errors in the client are silently ignored.

**Describe the solution you'd like**
Similarly to the api-server, it would be good if we could report errors to Sentry.

**Describe alternatives you've considered**
Nothing obvious leaps to mind - Sentry works well.

**Additional context**
We would need to be careful not to overwhelm the quota and only report things that are genuinely unexpected.  

There is already some code lying around from a previous attempt: https://github.com/freeCodeCamp/freeCodeCamp/issues/39574#issuecomment-701344881

### PR DESCRIPTION
**Pull request by TristanToye**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #39751

Got this to log to a free sentry account with a gitpod running. Followed similar pattern to server. I am unclear on if there is a better place to put this integration for the project setup. Also wondering if there is a known client side error I could trigger here more easily than creating a fake one for testing steps?

### CODE REVIEW

**Comment by raisedadead** [comment]
@ojeytonwilliams is there anything preventing us from getting this in?

**Comment by ojeytonwilliams** [comment]
No, I don't have any reservations. It just needs reviewing and the lock files updating (I'll handle that since we've left this sitting for so long).

**Comment by raisedadead** [comment]
Perfect. Thanks a lot.

File: client/src/html.tsx [hunk 1]
```diff
@@ -1,4 +1,18 @@
...
+
+if (sentry.dns === 'dsn_from_sentry_dashboard') {
+  console.log('Sentry reporting disabled unless DSN is provided!');
+} else {
+  Sentry.init({
+    dsn: sentry.dns,
+    integrations: [new Integrations.BrowserTracing()],
+    tracesSampleRate: 1.0
+  });
+  console.log('Sentry initialized');
```
Comment (nhcarrigan) [inline client/src/html.tsx:14]: Do we want to be using the same DSN for both the client and API?

Comment (ojeytonwilliams) [reply client/src/html.tsx:14]: Good catch. No, that would be very confusing.

I've set up a new project and I'll make that key available to the client.

File: client/src/utils/report-error.ts [hunk 2]
```diff
@@ -1,7 +1,13 @@
-// TODO: integrate with Sentry?
+import * as Sentry from '@sentry/react';
+import { sentry } from '../../../config/secrets';
+
 export function reportClientSideError(
   e: Error,
-  message = 'Unhandled error'
-): void {
-  return console.error(`Client: ${message}`, e);
+  message = sentry.dns === 'dsn_from_sentry_dashboard'
```
Comment (ShaunSHamilton) [inline client/src/utils/report-error.ts:6]: Can we fix this in `/config/secrets.js`? That is, make it `sentry.dsn`?

Comment (ojeytonwilliams) [reply client/src/utils/report-error.ts:6]: My typo has survived two years so far, but sure.

Comment (nhcarrigan) [reply client/src/utils/report-error.ts:6]: I propose `sentry.dsn$`

Comment (ojeytonwilliams) [reply client/src/utils/report-error.ts:6]:
```js
$(function() {
 Sentry.init({
    dsn: sentry.dsn$,
    integrations: [new Integrations.BrowserTracing()],
    tracesSampleRate: 1.0
  });
});
```

**Review by raisedadead — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM @TristanToye 🎉 I know this has been sitting here for a while, but thanks for working on this and congratulations on your first contribution to freeCodeCamp 🎉 

I suppose we need to add some keys to the builds - anyone want to volunteer?

**Comment by ojeytonwilliams** [comment]
> I suppose we need to add some keys to the builds - anyone want to volunteer?

Sure, I'll do that now.

**Pull request merged**